client/icecc-create-env.in

```diff
@@ -336,9 +337,24 @@ if test -f /etc/ld.so.conf; then
     fi
     echo "$directive $path"
   done </etc/ld.so.conf >$tmp_ld_so_conf
+
+  # Work around new directory layout for clang in ubuntu >= 16.10
+  if [ -n "$clangprefix" ]; then
+      echo "$clangprefix/lib" >> $tmp_ld_so_conf
+  fi
+
   add_file $tmp_ld_so_conf /etc/ld.so.conf
 fi
 
+for file in /etc/ld.so.conf.d/*.conf; do
+    if [ -e "$file" ]; then
+        add_file "$file"
+    else
+        # This can happen with a dangling symlink
+        echo "skipping non-existent file $file"
+    fi
+done
+
```

Comment:
Why would anything in the environment tarball need this? Is there anything in these extra files that's used by the compiler?

Reply:
At least on arch, ld.so.conf is just `include /etc/ld.so.conf.d/*.conf`, and everything is in the sub-confs.